R-package/tests/testthat/test_interaction_constraints.R

```diff
@@ -0,0 +1,34 @@
+require(xgboost)
+
+context("interaction constraints")
+
+set.seed(1024)
+x1 <- rnorm(1000, 1)
+x2 <- rnorm(1000, 1)
+x3 <- sample(c(1,2,3), size=1000, replace=TRUE)
+y <- x1 + x2 + x3 + x1*x2*x3 + rnorm(1000, 0.001) + 3*sin(x1)
+train <- matrix(c(x1,x2,x3), ncol = 3)
+colnames(train) <- c('x1','x2','x3')
+
+test_that("interaction constraints for regression", {
+  bst <- xgboost(data = train, label = y, max_depth = 3,
+                 eta = 0.1, nthread = 2, nrounds = 100, verbose = 0,
+                 int_constraints = list(c('x1','x2')))
+  
+	preds <- lapply(c(1,2,3), function(x){
+		tmat <- matrix(c(x1,x2,rep(x,1000)), ncol=3)
+		return(predict(bst, tmat))
+	})
+	
+	# Changing x3 values should have the same effect on all observations
```

Comment:
Why is this the case? You could have the root node split at `x3`, so in that case changing `x3` would have different effects on instances.

Reply:
I probably should have made the comment clearer.  If all observations started with the same value of x3 and we change x3 by the same amount, then the change in predictions should be the same across all observations - since x3 is constrained to be independent of x1 and x2 in the model.

Reply:
Ah it makes sense now. Can you clarify the comment a bit?

Reply:
sure, will update it